### PR TITLE
feat(v4): updated buttons in UI packages

### PR DESCRIPTION
**Pull request by aliemir**
- [x] deprecated `resourceNameOrRouteName` in buttons.
- [x] added `resource` property as a replacement, which is a better fit for consistency.
- [x] added codemod for the replacement
- [x] updated docs to reflect the changes.
- [x] added `meta` property to the docs of the buttons. (related to the router provider changes)

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: .changeset/seven-wolves-wait.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@pankod/refine-codemod": major
```
Comment (salihozdemir) [inline .changeset/seven-wolves-wait.md:2]:
```suggestion
"@pankod/refine-codemod": minor
```

It should be minor, shouldn't it?

File: packages/antd/src/components/buttons/show/index.tsx [hunk 2]
```diff
@@ -64,9 +67,12 @@ export const ShowButton: React.FC<ShowButtonProps> = ({
     };
 
     const showUrl =
-        (resource || propResourceNameOrRouteName) && (recordItemId || id)
+        (resource || resourceNameFromProps || propResourceNameOrRouteName) &&
```
Comment (salihozdemir) [inline packages/antd/src/components/buttons/show/index.tsx:70]: Is `resource` created based on `resourceNameFromProps` and `propResourceNameOrRouteName`? I mean, `useResource` hook returns `resource` according to its params. In this case, the params are `resourceNameFromProps` and `propResourceNameOrRouteName`.

File: packages/antd/src/components/buttons/list/index.tsx [hunk 3]
```diff
@@ -42,7 +43,9 @@ export const ListButton: React.FC<ListButtonProps> = ({
 
     const translate = useTranslate();
 
-    const { resource } = useResource(propResourceNameOrRouteName);
+    const { resource } = useResource(
```
Comment (alicanerdurmaz) [inline packages/antd/src/components/buttons/list/index.tsx:46]: `pickDeprecated` 😞  ?

File: packages/antd/src/components/buttons/list/index.tsx [hunk 4]
```diff
@@ -100,12 +108,15 @@ export const ListButton: React.FC<ListButtonProps> = ({
                     (children ??
                         translate(
                             `${
-                                resource?.name ?? propResourceNameOrRouteName
+                                resource?.name ??
```
Comment (alicanerdurmaz) [inline packages/antd/src/components/buttons/list/index.tsx:111]: `pickDeprecated` 😞  ?

**Review by alicanerdurmaz — DISMISSED** [review]

**Review by yildirayunlu — DISMISSED** [review]

File: packages/antd/src/components/buttons/show/index.tsx (hunk 2 above)
Comment (aliemir) [reply packages/antd/src/components/buttons/show/index.tsx]: updated

File: packages/antd/src/components/buttons/list/index.tsx (hunk 4 above)
Comment (aliemir) [reply packages/antd/src/components/buttons/list/index.tsx:104]: updated

**Review by BatuhanW — APPROVED** [review]

**Review by alicanerdurmaz — APPROVED** [review]

**Pull request merged**